lib/rubocop/cop/lint/void.rb

```diff
@@ -124,7 +124,10 @@ def check_void_expression(node)
         end
 
         def check_nonmutating(node)
-          return unless node.send_type? && NONMUTATING_METHODS.include?(node.method_name)
+          right_node_type = (node.send_type? || node.block_type? || node.numblock_type?)
+          right_method = NONMUTATING_METHODS.include?(node.method_name)
+
+          return unless right_node_type && right_method
```

Comment:
It seems that there were redundant conditions. So this is probably better:

```suggestion
          return unless NONMUTATING_METHODS.include?(node.method_name)
```